File: fibonacci/formula.cpp
#include <bits/stdc++.h>

const double sqrt_5 = sqrt(5);
const double phi = (1. + sqrt_5) / 2.;
const double phi_conjugate = (1. - sqrt_5) / 2.;

unsigned long long fibonacci_formula(int n) {
    return floor((pow(phi ,n) - pow(phi_conjugate ,n)) / sqrt_5);
}

File: fibonacci/formula.py
import math

SQRT_5 = math.sqrt(5.)
PHI = (1. + SQRT_5) / 2.
PHI_CONJUGATE = (1. - SQRT_5) / 2.

def fibonacci_formula(n):
    return math.floor((math.pow(PHI, n) - math.pow(PHI_CONJUGATE, n)) / SQRT_5)
